File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DataGridView/Programming/CellFormatingPage.vb
Imports System.Drawing

Namespace CompanionKit.Controls.DataGridView.Programming

	Public Class CellFormatingPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()

		End Sub

        Private Sub CellFormatingPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            'Fill up colors ComboBox with known Colors.
            Dim marrKnownColors As KnownColor() = DirectCast([Enum].GetValues(GetType(KnownColor)), KnownColor())
            Dim marrColors As Color() = New Color(marrKnownColors.Length - 1) {}
            For i As Integer = 0 To marrKnownColors.Length - 1
                marrColors(i) = Color.FromKnownColor(marrKnownColors(i))
            Next

            Me.mobjForeColorComboBox.DataSource = marrColors
            Me.mobjForeColorComboBox.ColorMember = "Name"
            Me.mobjForeColorComboBox.DisplayMember = "Name"

            Me.mobjBackColorComboBox.DataSource = marrColors.Clone()
            Me.mobjBackColorComboBox.ColorMember = "Name"
            Me.mobjBackColorComboBox.DisplayMember = "Name"

            'Fill up DataGridView
            For i As Integer = 1 To 19
                Me.mobjUserDS.Users.AddUsersRow(String.Format("User{0}", i), String.Format("user{0}@gmail.com", i), String.Format("8-800-236589{0}", i), "Franklin", String.Format("10{0} Murfreeboro Rd.", i), "USA", _
                 "Tennessee", "37064")
            Next


        End Sub

        ''' <summary>
        ''' Handles CellFormatting event for a DataGridView.
        ''' Updates text of the informed label with names of new colors.
        ''' </summary>
        Private Sub mobjDataGridView_CellFormatting(sender As Object, e As DataGridViewCellFormattingEventArgs) Handles mobjDataGridView.CellFormatting
            ' Update text of the informed label with names of new colors.
            If (e.RowIndex > -1 AndAlso e.RowIndex < Me.mobjDataGridView.Rows.Count) AndAlso (e.ColumnIndex > -1 AndAlso e.ColumnIndex < mobjDataGridView.Columns.Count) Then
                Dim selectedCell As DataGridViewCell = Me.mobjDataGridView.Rows(e.RowIndex).Cells(e.ColumnIndex)

                Me.mobjInformLabel.Text = String.Format("Cell[{0},{1}] has foreground {2} and background {3} colors.", e.RowIndex, e.ColumnIndex, IIf(Not selectedCell.Style.ForeColor.IsEmpty, selectedCell.Style.ForeColor.Name, Me.mobjDataGridView.DefaultCellStyle.ForeColor.Name), IIf(Not selectedCell.Style.BackColor.IsEmpty, selectedCell.Style.BackColor.Name, Me.mobjDataGridView.DefaultCellStyle.BackColor.Name))
            End If
        End Sub

        ''' <summary>
        ''' Handles SelectedIndexChanged event for ComboBox.
        ''' Changes foreground color for selected cells.
        ''' </summary>
        Private Sub mobjForeColorComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjForeColorComboBox.SelectedIndexChanged

            ' Changes foreground color for selected cells.
            For i As Integer = 0 To Me.mobjDataGridView.SelectedCells.Count - 1
                Dim mobjCell As DataGridViewCell = Me.mobjDataGridView.SelectedCells(i)
                mobjCell.Style.ForeColor = DirectCast(Me.mobjForeColorComboBox.SelectedItem, Color)
            Next
        End Sub

        ''' <summary>
        ''' Handles SelectedIndexChanged event for ComboBox.
        ''' Changes background color for selected cells.
        ''' </summary>
        Private Sub mobjBackColorComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjBackColorComboBox.SelectedIndexChanged
            ' Change background color for selected cells.
            For i As Integer = 0 To Me.mobjDataGridView.SelectedCells.Count - 1
                Dim mobjCell As DataGridViewCell = Me.mobjDataGridView.SelectedCells(i)
                mobjCell.Style.BackColor = DirectCast(Me.mobjBackColorComboBox.SelectedItem, Color)
            Next
        End Sub

        ''' <summary>
        ''' Handles SelectionChanged event for a DataGridView.
        ''' Selects color in the ComboBox appropriate to colors of cell.
        ''' </summary>
        Private Sub mobjDataGridView_SelectionChanged(sender As Object, e As EventArgs) Handles mobjDataGridView.SelectionChanged
            If Me.mobjDataGridView.SelectedCells.Count > 0 Then
                Dim mobjFirstCell As DataGridViewCell = Me.mobjDataGridView.SelectedCells(0)

                ' Update colors ComboBox
                ' Enable color ComboBoxes only when DataGridView cell has selected and before anything cells were not selected.
                If Not Me.mobjBackColorComboBox.Enabled Then
                    Me.mobjBackColorComboBox.Enabled = True
                End If
                If Not Me.mobjForeColorComboBox.Enabled Then
                    Me.mobjForeColorComboBox.Enabled = True
                End If
                mobjForeColorComboBox.SelectedItem = IIf(Not mobjFirstCell.Style.ForeColor.IsEmpty, mobjFirstCell.Style.ForeColor, Me.mobjDataGridView.DefaultCellStyle.ForeColor)
                mobjBackColorComboBox.SelectedItem = IIf(Not mobjFirstCell.Style.BackColor.IsEmpty, mobjFirstCell.Style.BackColor, Me.mobjDataGridView.DefaultCellStyle.BackColor)
            Else
                Me.mobjBackColorComboBox.Enabled = False

                Me.mobjForeColorComboBox.Enabled = False
            End If
        End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DataGridView/Programming/CellFormatingPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.DataGridView.Programming
{
    public partial class CellFormatingPage : UserControl
    {

        public CellFormatingPage()
        {
            InitializeComponent();
        }

        private void CellFormatingPage_Load(object sender, EventArgs e)
        {
            //Fill up colors ComboBox with known Colors.
            KnownColor[] marrKnownColors = (KnownColor[])Enum.GetValues(typeof(KnownColor));
            Color[] marrColors = new Color[marrKnownColors.Length];
            for (int i = 0; i < marrKnownColors.Length; ++i)
            {
                marrColors[i] = Color.FromKnownColor(marrKnownColors[i]);
            }

            this.mobjForeColorComboBox.DataSource = marrColors;
            this.mobjForeColorComboBox.ColorMember = "Name";
            this.mobjForeColorComboBox.DisplayMember = "Name";

            this.mobjBackColorComboBox.DataSource = marrColors.Clone();
            this.mobjBackColorComboBox.ColorMember = "Name";
            this.mobjBackColorComboBox.DisplayMember = "Name";

            //Fill up DataGridView
            for (int i = 1; i < 20; ++i)
            {
                this.mobjUserDS.Users.AddUsersRow(string.Format("User{0}", i), 
                                         string.Format("user{0}@gmail.com", i),
                                         string.Format("8-800-236589{0}", i), "Franklin",
                                         string.Format("10{0} Murfreeboro Rd.", i), "USA", 
                                         "Tennessee", "37064");
            }
           

        }

        /// <summary>
        /// Handles CellFormatting event for a DataGridView.
        /// Updates text of the informed label with names of new colors.
        /// </summary>
        private void mobjDataGridView_CellFormatting(object sender, DataGridViewCellFormattingEventArgs e)
        {
            // Update text of the informed label with names of new colors.
            if ((e.RowIndex > -1 && e.RowIndex < this.mobjDataGridView.Rows.Count)
                && (e.ColumnIndex > -1 && e.ColumnIndex < mobjDataGridView.Columns.Count))
            {
                DataGridViewCell selectedCell = this.mobjDataGridView.Rows[e.RowIndex].Cells[e.ColumnIndex];

                this.mobjInformLabel.Text = string.Format("Cell[{0},{1}] has foreground {2} and background {3} colors.",
                                                    e.RowIndex, e.ColumnIndex, 
                                                    !selectedCell.Style.ForeColor.IsEmpty ? selectedCell.Style.ForeColor.Name : this.mobjDataGridView.DefaultCellStyle.ForeColor.Name,
                                                    !selectedCell.Style.BackColor.IsEmpty ? selectedCell.Style.BackColor.Name : this.mobjDataGridView.DefaultCellStyle.BackColor.Name);
            }
        }

        /// <summary>
        /// Handles SelectedIndexChanged event for ComboBox.
        /// Changes foreground color for selected cells.
        /// </summary>
        private void mobjForeColorComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {

            // Changes foreground color for selected cells.
            for (int i = 0; i < this.mobjDataGridView.SelectedCells.Count; ++i)
            {
                DataGridViewCell mobjCell = this.mobjDataGridView.SelectedCells[i];
                mobjCell.Style.ForeColor = (Color)this.mobjForeColorComboBox.SelectedItem;
            }
        }

        /// <summary>
        /// Handles SelectedIndexChanged event for ComboBox.
        /// Changes background color for selected cells.
        /// </summary>
        private void mobjBackColorComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            // Change background color for selected cells.
            for (int i = 0; i < this.mobjDataGridView.SelectedCells.Count; ++i)
            {
                DataGridViewCell mobjCell = this.mobjDataGridView.SelectedCells[i];
                mobjCell.Style.BackColor = (Color)this.mobjBackColorComboBox.SelectedItem;
            }
        }

        /// <summary>
        /// Handles SelectionChanged event for a DataGridView.
        /// Selects color in the ComboBox appropriate to colors of cell.
        /// </summary>
        private void mobjDataGridView_SelectionChanged(object sender, EventArgs e)
        {
            if (this.mobjDataGridView.SelectedCells.Count > 0)
            {
                DataGridViewCell mobjFirstCell = this.mobjDataGridView.SelectedCells[0];

                // Update colors ComboBox
                // Enable color ComboBoxes only when DataGridView cell has selected and before anything cells were not selected.
                if (!this.mobjBackColorComboBox.Enabled)
                {
                    this.mobjBackColorComboBox.Enabled = true;
                }
                if (!this.mobjForeColorComboBox.Enabled)
                {
                    this.mobjForeColorComboBox.Enabled = true;
                }
                mobjForeColorComboBox.SelectedItem = !mobjFirstCell.Style.ForeColor.IsEmpty ? mobjFirstCell.Style.ForeColor : this.mobjDataGridView.DefaultCellStyle.ForeColor;
                mobjBackColorComboBox.SelectedItem = !mobjFirstCell.Style.BackColor.IsEmpty ? mobjFirstCell.Style.BackColor : this.mobjDataGridView.DefaultCellStyle.BackColor;
            }
            else
            {
                this.mobjBackColorComboBox.Enabled = false;
                this.mobjForeColorComboBox.Enabled = false;
               
            }
        }

    }
}